### PR TITLE
Revert PEP validation back to python samplesheet check

### PR DESCRIPTION
**Pull request by jfy133**
PEP/Eido is unfortunately not ready (it doesn't yet provide descriptive error messages).

Closes https://github.com/nf-core/taxprofiler/issues/173

Note, I'm going to test against a few broken samplesheets and see that the python script still covers everything we need

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/taxprofiler/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/taxprofiler _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by jfy133** [comment]
All broken samplesheets are here here attached in [broken_samplesheets.zip](https://github.com/nf-core/taxprofiler/files/10205654/broken_samplesheets.zip)


With samplesheet_broken_missingcolumn.csv:

```
  Traceback (most recent call last):
    File "/home/jfellows/Documents/git/nf-core/taxprofiler/bin/check_samplesheet.py", line 236, in <module>
      sys.exit(main())
    File "/home/jfellows/Documents/git/nf-core/taxprofiler/bin/check_samplesheet.py", line 232, in main
      check_samplesheet(args.FILE_IN, args.FILE_OUT)
    File "/home/jfellows/Documents/git/nf-core/taxprofiler/bin/check_samplesheet.py", line 111, in check_samplesheet
      lspl = [line_parsed[i] for i in header_locs.values()]
    File "/home/jfellows/Documents/git/nf-core/taxprofiler/bin/check_samplesheet.py", line 111, in <listcomp>
      lspl = [line_parsed[i] for i in header_locs.values()]
  IndexError: list index out of range
```

From samplesheet_broken_wrongplatform.csv

```
  Traceback (most recent call last):
    File "/home/jfellows/Documents/git/nf-core/taxprofiler/bin/check_samplesheet.py", line 236, in <module>
      sys.exit(main())
    File "/home/jfellows/Documents/git/nf-core/taxprofiler/bin/check_samplesheet.py", line 232, in main
      check_samplesheet(args.FILE_IN, args.FILE_OUT)
    File "/home/jfellows/Documents/git/nf-core/taxprofiler/bin/check_samplesheet.py", line 175, in check_samplesheet
      print_error(
  TypeError: print_error() takes from 1 to 3 positional arguments but 4 were given
```

all other tests pass

File: bin/check_samplesheet.py
Comment (jfy133) [reply bin/check_samplesheet.py:186]: These are already accounted for in the FASTP/AdapterRemoval modules

Comment (jfy133) [reply bin/check_samplesheet.py:186]: Sorry to clarify: Illumina single-end short reads are already accounted for. I don't have experience with Ion Torrent data, so I don't know if those modules would work with those 'out of the box'

Comment (sofstam) [reply bin/check_samplesheet.py:186]: Ok if they are accounted for in the fastp/adapterremoval modules

**Review by sofstam — APPROVED** [review]
I do not have any other comments, lgtm.

File: subworkflows/local/input_check.nf
Comment (Midnighter) [inline subworkflows/local/input_check.nf:29]: I prefer assignment to using `.set` but either is valid code, of course.

File: subworkflows/local/input_check.nf
Comment (Midnighter) [inline subworkflows/local/input_check.nf:37]: Should we return only one element here? Will be lots of duplicated information.

```suggestion
    versions = SAMPLESHEET_CHECK.out.versions.first()  // channel: [ versions.yml ]
```

**Review by Midnighter — COMMENTED** [review]
First review looks good. I'll look at those specific errors now.

**Comment by Midnighter** [comment]
Seems like we're using a very old version of the samplesheet Python script, too.

File: bin/check_samplesheet.py
Comment (Midnighter) [inline bin/check_samplesheet.py:177]: Needs to be a consecutive string.

```suggestion
                        f"Instrument platform {instrument_platform} is not supported! "
                        f"List of supported platforms {', '.join(INSTRUMENT_PLATFORMS)}",
```

File: bin/check_samplesheet.py
Comment (Midnighter) [inline bin/check_samplesheet.py:126]: Need to check first that the expected number of columns is present. Maybe something like the following will work.

```suggestion

            # Check valid number of columns per row
            if len(line_parsed) < len(HEADER):
                print_error(
                    "Invalid number of columns (minimum = {})!".format(len(HEADER)),
                    "Line",
                    line,
                )
            num_cols = len([x for x in line_parsed if x])
            if num_cols < MIN_COLS:
                print_error(
                    "Invalid number of populated columns (minimum = {})!".format(MIN_COLS),
                    "Line",
                    line,
                )
                
            lspl = [line_parsed[i] for i in header_locs.values()]
```

**Review by Midnighter — CHANGES_REQUESTED** [review]
I think the last to comments should address the errors.

File: bin/check_samplesheet.py
Comment (jfy133) [reply bin/check_samplesheet.py:60]: We are generally... yes sure. Why not. If someone complains we can 'undo' it.

I guess my main thinking of supporting at least the FASTA is if people want to run genomes/contigs through it which often aren't compressed for microbial things but :shrug:

File: subworkflows/local/input_check.nf
Comment (jfy133) [reply subworkflows/local/input_check.nf:29]: Agreed! I have a feeing this is very old what I'm putting back in :sweat_smile:

**Comment by jfy133** [comment]
I can confirm all the changes work as expected @Midnighter ! Useful errors for all.

I'll wait for tests passing _or_ your approval whatever comes first (before merging - we need this to make the template merge easier :+1: )

File: bin/check_samplesheet.py
Comment (Midnighter) [reply bin/check_samplesheet.py]: Fine with me to keep it. I think the pipeline should succeed with uncompressed input.

**Review by Midnighter — APPROVED** [review]
All good from my end.

**Comment by jfy133** [comment]
All the failures are API limits from downloading nextflow but otherwise pass, so will merge!

**Pull request merged**
